extensions/ql-vscode/src/vscode-tests/no-workspace/remote-query-history.test.ts

```diff
@@ -0,0 +1,342 @@
+    await qhm.readQueryHistory();
+    const addSpy = sandbox.spy();
+    disposables.push(qhm.onDidAddQueryItem(addSpy));
+    const removeSpy = sandbox.spy();
+    disposables.push(qhm.onDidRemoveQueryItem(removeSpy));
+
+    // Remove the first query
+    await qhm.handleRemoveHistoryItem(qhm.treeDataProvider.allHistory[0]);
+    expect(removeSpy.getCall(0).args[0]).to.deep.eq(rawQueryHistory[0]);
+    expect(removeSpy.callCount).to.eq(1);
+    expect(addSpy.callCount).to.eq(0);
+    expect(qhm.treeDataProvider.allHistory).to.deep.eq(rawQueryHistory.slice(1));
+
+    // Add it back
+    qhm.addQuery(rawQueryHistory[0]);
+    expect(removeSpy.callCount).to.eq(1);
+    expect(addSpy.getCall(0).args[0]).to.deep.eq(rawQueryHistory[0]);
+    expect(addSpy.callCount).to.eq(1);
+    expect(qhm.treeDataProvider.allHistory).to.deep.eq([rawQueryHistory[1], rawQueryHistory[0]]);
+  });
+
+  it('should remove two queries from history', async () => {
+    await qhm.readQueryHistory();
+    const addSpy = sandbox.spy();
+    disposables.push(qhm.onDidAddQueryItem(addSpy));
+    const removeSpy = sandbox.spy();
+    disposables.push(qhm.onDidRemoveQueryItem(removeSpy));
+
+    // Remove the both queries
+    // Just for fun, let's do it in reverse order
+    await qhm.handleRemoveHistoryItem(undefined!, [qhm.treeDataProvider.allHistory[1], qhm.treeDataProvider.allHistory[0]]);
+    expect(removeSpy.getCall(0).args[0]).to.deep.eq(rawQueryHistory[1]);
+    expect(removeSpy.getCall(1).args[0]).to.deep.eq(rawQueryHistory[0]);
+    expect(qhm.treeDataProvider.allHistory).to.deep.eq([]);
+    expect(removeSpy.callCount).to.eq(2);
+
+    // also, both queries should be removed from on disk storage
+    expect(fs.readJSONSync(path.join(STORAGE_DIR, 'workspace-query-history.json'))).to.deep.eq({
+      version: 1,
+      queries: []
+    });
+  });
+
+  it('should handle a click', async () => {
+    await qhm.readQueryHistory();
+    const openSpy = sandbox.spy();
+    disposables.push(qhm.onWillOpenQueryItem(openSpy));
+
+    await qhm.handleItemClicked(qhm.treeDataProvider.allHistory[0], []);
+    expect(openSpy.getCall(0).args[0]).to.deep.eq(rawQueryHistory[0]);
+  });
+
+  it('should get the query text', async () => {
+    await qhm.readQueryHistory();
+    await qhm.handleShowQueryText(qhm.treeDataProvider.allHistory[0], []);
+
+    expect(showTextDocumentSpy).to.have.been.calledOnce;
+    expect(openTextDocumentSpy).to.have.been.calledOnce;
+
+    const uri: Uri = openTextDocumentSpy.getCall(0).args[0];
+    expect(uri.scheme).to.eq('codeql');
+    const params = new URLSearchParams(uri.query);
+    expect(params.get('isQuickEval')).to.eq('false');
+    expect(params.get('queryText')).to.eq(rawQueryHistory[0].remoteQuery.queryText);
+  });
+
+  describe('AnalysisResultsManager', () => {
+
+    let mockCredentials: any;
+    let mockOctokit: any;
+    let mockLogger: any;
+    let arm: AnalysesResultsManager;
+
+    beforeEach(() => {
+      mockOctokit = {
```

Comment:
I'm a little surprised we have to mock at this level and not at `gh-api-client` level which is what the `AnalysesResultsManager` is using.. Is this because we want to test as closer to the "edge" as possible?

Reply:
Oh sorry I didn't realise that was the case. That's a bit yucky, but appreciate it would be more work at this point!